pcp/versions/2.0/session.md

```diff
@@ -0,0 +1,48 @@
+Sessions
+===
+
+PCP clients must establish a session with a broker before they can use
+its capabilities. A session is an authenticated connection between a client and
+broker and the broker associating a [client URI][1] with that connection.
+
+This is done by appending the *client_type* as a path element to the underlying
+connection URI. If the resulting URI is not authorized by the broker, the
+connection will be closed.
```

Comment:
It is. Something like `(.. ws getSession getUpgradeRequest getParameterMap)`. I preferred the path as it gives us more flexibility internally. We can either write separate handlers for the client types we want to support, or parse the path to get the client type.

Query parameters do seem clearer though, and that makes implementing a case statement based on them more straight-forward.